MEP_Engine/Modify/Transform.cs

```diff
@@ -26,15 +26,39 @@
 using BH.oM.MEP.System;
 using BH.oM.Reflection.Attributes;
 using System.ComponentModel;
+using System.Linq;
+using BH.oM.MEP.System.Fittings;
 
 namespace BH.Engine.MEP
 {
     public static partial class Modify
     {
         /***************************************************/
-        /**** Interface Methods - IElements             ****/
+        /****               Public Methods              ****/
         /***************************************************/
 
+        [Description("Transforms the Fitting's location and connections location points by the transform matrix. Only rigid body transformations are supported.")]
+        [Input("fitting", "Fitting to transform.")]
+        [Input("transform", "Transform matrix.")]
+        [Input("tolerance", "Tolerance used in the check whether the input matrix is equivalent to the rigid body transformation.")]
+        [Output("transformed", "Modified Fitting with unchanged properties, but transformed location and connections location points.")]
+        public static Fitting Transform(this Fitting fitting, TransformMatrix transform, double tolerance = Tolerance.Distance)
+        {
+            if (!transform.IsRigidTransformation(tolerance))
+            {
+                BH.Engine.Reflection.Compute.RecordError("Transformation failed: only rigid body transformations are currently supported.");
+                return null;
+            }
+
+            Fitting result = fitting.GetShallowClone() as Fitting;
```

Comment:
```suggestion
            Fitting result = fitting.ShallowClone();
```

Based on the work @alelom is currently doing, it's probably best not to put in another `GetShallowClone` when he's removing them. @alelom can you confirm this is the right suggestion for correction based on your current work?

Reply:
Yes definitely! Please don't give me another one to remove! 🤣